src/client/tls_stream.rs

```diff
@@ -42,3 +42,30 @@ pub(crate) async fn create_tls_stream<S: AsyncRead + AsyncWrite + Unpin + Send>(
 ) -> crate::Result<TlsStream<S>> {
     opentls_tls_stream::create_tls_stream(config, stream).await
 }
+
+#[cfg(any(feature = "native-tls", feature = "vendored-openssl"))]
+mod iter_certs {
+    const BEGIN: &[u8] = b"-----BEGIN";
+    pub struct IterCertBundle<'a> {
+        bundle: &'a [u8],
+        current_begin: Option<usize>,
+    }
+    impl<'a> IterCertBundle<'a> {
+        pub fn new(bundle: &'a [u8]) -> Self {
+            Self {
+                bundle,
+                current_begin: bundle.windows(BEGIN.len()).position(|x| x == BEGIN),
+            }
+        }
+    }
+    impl<'a> Iterator for IterCertBundle<'a> {
+        type Item = &'a [u8];
+        fn next(&mut self) -> Option<&'a [u8]> {
+            self.current_begin.map(|begin| {
+                let next_begin = self.bundle.windows(BEGIN.len()).skip(begin + 1).position(|x| x == BEGIN).map(|x| x + begin + 1);
+                self.current_begin = next_begin;
+                &self.bundle[begin..next_begin.unwrap_or(self.bundle.len() - 1)]
+            })
+        }
+    }
```

Comment:
This looks good, but I think it ignores the markers for the end of the certificate. Have you tested this outside of this repo? Do you think a test case in this repo would be doable, to confirm that this doesn't break?

Reply:
>I think it ignores the markers for the end of the certificate

I'm not sure I understand. In any valid PEM bundle, each certificate is terminated before the next one begins. The only thing this code does is splitting a bundle of concatenated PEM structures into individual PEM structures. The PEM format explicitly allows arbitrary data (which parsers must skip over) before the BEGIN and after the END marker, so cleanly terminating after the END marker is not a requirement.

>Have you tested this outside of this repo

We're using this code in production.

>Do you think a test case in this repo would be doable

Sure. Do you think it would be sufficient to just have a small unit test that loads an arbitrary bundle using `include_str!()` and verifies that the expected number of certificates is produced? Or did you have something else in mind?